cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java

```diff
@@ -86,21 +84,70 @@ public void create(NamespaceMeta namespaceMeta) throws IOException, ExploreExcep
    */
   @Override
   public void delete(NamespaceId namespaceId) throws IOException, ExploreException, SQLException {
+
+    deleteLocation(namespaceId);
+
+    if (cConf.getBoolean(Constants.Explore.EXPLORE_ENABLED)) {
+      exploreFacade.removeNamespace(namespaceId.toId());
+    }
+  }
+
+  private void deleteLocation(NamespaceId namespaceId) throws IOException {
     // TODO: CDAP-1581: Implement soft delete
     Location namespaceHome = namespacedLocationFactory.get(namespaceId.toId());
-    if (namespaceHome.exists()) {
-      if (!namespaceHome.delete(true)) {
-        throw new IOException(String.format("Error while deleting home directory '%s' for namespace '%s'",
-                                            namespaceHome, namespaceId.getNamespace()));
+    try {
+      if (hasCustomLocation(namespaceQueryAdmin.get(namespaceId.toId()))) {
+        LOG.debug("Custom location mapping %s was found while deleting namespace %s. Skipping location delete.",
+                  namespaceHome, namespaceId);
+      } else {
+        // a custom location was not provided for this namespace so cdap is responsible for managing the lifecycle of
+        // the location hence delete it.
+        if (namespaceHome.exists() && !namespaceHome.delete(true)) {
+          throw new IOException(String.format("Error while deleting home directory '%s' for namespace '%s'",
+                                              namespaceHome, namespaceId));
+        }
       }
-    } else {
-      // warn that namespace home was not found and skip delete step
-      LOG.warn(String.format("Home directory '%s' for namespace '%s' does not exist.",
-                             namespaceHome, namespaceId));
+    } catch (Exception e) {
+      throw new IOException(String.format("Error while deleting home directory %s for namespace %s ", namespaceHome,
```

Comment:
sometimes you surround the %s with single quotes, sometimes you don't. is that on purpose?